### DISCUSSION TITLE
Setting default value for form element which make sense for their type

### DISCUSSION THREAD

**Discussion by sujeet-agrahari** [react-hook-form/react-hook-form#8252] [Q&A]
I am using `react-hook-form` with `mui`. Many times I get below error,
> Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component.

I understand the error. If I don't pass any default value which result in `undefined` so react treats it as uncontrolled component.

But later `react-hook-form` tries to control it by setting the value, hence the error.

Setting defaultValue for `TextField` with text type to `""` could be accepted.

But what to set default value for numbers, let's say for field `totalFees`? `0`? I don't think so.

Here is the questions,

1. Should I always pass default value to each `Controller` when I use `react-hook-form` with `Mui`?

2. What default value should be set for types i.e. text, number, date etc.

3. Should I set default value on Controller or rendered input element i.e. TextField?

**Comment by ritikbanger** [comment]
This behaviour is due to the fact that at some point your MUI component is behaving like uncontrolled input and at some point it becomes controlled input.

If you are using controller with MUI component, it is recommended to give default value.

Please share a codesandbox for more clarity on what actually you are facing.
